app/main.py

```diff
@@ -1,8 +1,29 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people_list: list[dict]) -> list:
+    add_person(people_list)
+    set_spouses(people_list)
+    return list(Person.people.values())
+
+
+def add_person(people_list: list[dict]) -> None:
+    for person in people_list:
+        name = person["name"]
+        age = person["age"]
+        Person(name, age)
+
+
+def set_spouses(people_list: list[dict]) -> None:
+    for person in people_list:
+        name = person["name"]
+        if "wife" in person and person["wife"]:
+            Person.people[name].wife = Person.people[person["wife"]]
```

Comment:
You could make this code more efficient using walrus operator to not retrieve value by key several times
```suggestion
        person_instance = Person.people[person["name"]]
        if wife_name := person.get("wife"):
            person_instance.wife = Person.people[wife_name]
```
Apply the same change for husband as well

Reply:
fixed